pkg/routeagent/controllers/route/route.go

```diff
@@ -260,6 +264,40 @@ func (r *Controller) runPodWorker() {
 	}
 }
 
+func (r *Controller) overlappingSubnets(remoteSubnets []string) bool {
+	// If the remoteSubnets [*] overlap with local cluster Pod/Service CIDRs we
+	// should not update the IPTable rules on the host, as it will disrupt the
+	// functionality of the local cluster. So, lets validate that subnets do not
+	// overlap before we program any IPTable rules on the host for inter-cluster
+	// traffic.
+	// [*] Note: In a non-GlobalNet deployment, remoteSubnets will be a list of
+	// Pod/Service CIDRs, whereas in a GlobalNet deployment, it will be a list of
+	// globalCIDRs allocated to the clusters.
+	for _, serviceCidr := range r.localServiceCidr {
+		overlap, err := util.IsOverlappingCIDR(remoteSubnets, serviceCidr)
+		if err != nil {
+			// Ideally this case will never hit, as the subnets are valid CIDRs
+			klog.Warningf("unable to validate overlapping Service CIDR: %s", err)
+		}
+		if overlap {
+			klog.Errorf("Local Service CIDR %q, overlaps with remote cluster %s", serviceCidr, err)
```

Comment:
Would be good to add something about enabling globalnet to enable such a use case.

Reply:
Although we are currently logging this message, there is a separate issue to report the status via Operator (or for user to consume). I feel that the current log message is good enough because we would also have the following situation (an example)

We have three clusters - all non-overlapping CIDRs and user accidentally tries to join a fourth cluster that overlaps with one of the existing clusters. The objective of this PR is to avoid making changes to local cluster that disrupts the cluster connectivity. Whether user wants to delete submariner from all the existing clusters and then redeploy with GlobalNet or if he wants to update the CIDRs of fourth cluster (to avoid overlaps) is up to him.